src/Internal/Service/PaymentProcessingService.php

```diff
@@ -64,11 +64,11 @@ public function process_payment( int $order_id, bool $manual_capture = false ) {
 		// Start with a basis context.
 		$context = $this->create_payment_context( $order_id, $manual_capture );
 
-		$request         = new PaymentRequest( $this->legacy_proxy );
-		$initial_state   = $this->state_factory->create_state( InitialState::class, $context );
-		$completed_state = $initial_state->process( $request );
+		$request       = new PaymentRequest( $this->legacy_proxy );
+		$initial_state = $this->state_factory->create_state( InitialState::class, $context );
+		$new_state     = $initial_state->process( $request );
```

Comment:
Warning: @zmaglica's PR already returns `$initial_state->process()`, there will be a conflict here.

Reply:
Yes, I expect that this will happen, and ready to fix the conflict.